src/shapes/ellipse.class.ts

```diff
@@ -27,8 +27,6 @@ export class Ellipse extends FabricObject {
    */
   constructor(options: Record<string, unknown>) {
     super(options);
-    this.set('rx', (options && options.rx) || 0);
-    this.set('ry', (options && options.ry) || 0);
```

Comment:
I am surprised this doesn't break width and height values

Reply:
because constructor({...}) is doing that up in the chain with setOptions.
The issue could be if you pass with after rx,ry. But those are exactly the kind of things that needs to addressed when we setup the defaults.
I would also not bother, i prefer to say you have to set both of them, and zero is not a good value if not for both. I wouldn't try autofixing for the developer that have to understand this once.

Same issue with polygon bounding box, line points, radius and witdh/height. there are so many things we try to take care of we probably shouldn't.

If it wasn't late now i woudl say that width and height should be internal prop for geometry and the one we set for rect and text should be some sort of public width/height that nothing have to do with the geometry calculcations.

Reply:
> If it wasn't late now i woudl say that width and height should be internal prop for geometry and the one we set for rect and text should be some sort of public width/height that nothing have to do with the geometry calculcations.

Sounds good

> Same issue with polygon bounding box, line points, radius and witdh/height. there are so many things we try to take care of we probably shouldn't.

we should draft topics for a team meeting

Reply:
yes this whole how to use set/ set and _set, side effects, will be a recurring theme